File: systempra.m
function sys = systempra % Set system parameters, the unit is metre
sys.wl = 800e-9; % Free space wavelength of the laser beam
sys.re = 210; % Resolution in the image plane (number of pixels)
sys.r = 1e-6; % Calculation diameter in the image plane 
sys.z = 1e-6; % Calculation distance in the axial direction
sys.z0 = 0e-6; % Position of image plane
sys.NA = 1.4; % Numerical aperture of the objective
sys.n = 1.514; % Index of the immersion material 
               % For a water lens "n = 1.33"
               % For a air lens "n = 1"
sys.P = 'Circular';% Define the polarisation state of the incidence, which is case sensitive. 
                 % 'Linear' for linear polarisation
                 % 'Circular' for left-handed circular polarisation (LCP)
                 % 'Radial' for radial polarisation
                 % 'Azimuthal' for azimuthal polarisation
sys.e = 0; % Normalised radius of the annular aperture (from 0 to 1), "0" means a circular aperture
sys.m = 0; % Topological charge of the phase vortices("+" means left-handed phase vortices,
           % "-" meas right-handed phase vortices), "0" means a planewave
           % incident beam
sys.plane = 'R-Z';%'X-Y' for x-y plane calculation (case sensitive)
                  %'R-Z' for r-z plane calculaiton
end

File: python/systempra.py
"""
System parameters for the vectorial Debye focal field calculation.
Converted from systempra.m
"""


def systempra():
    """Return default system parameters as a dictionary."""
    return {
        'wl':    800e-9,    # Free-space wavelength (m)
        're':    210,       # Resolution (pixels per side)
        'r':     1e-6,      # XY calculation half-width (m)
        'z':     1e-6,      # Z calculation half-width (m)
        'z0':    0.0,       # Image plane axial position (m)
        'NA':    1.4,       # Numerical aperture
        'n':     1.514,     # Refractive index  (oil=1.514, water=1.33, air=1.0)
        'P':     'Circular',# Polarisation: 'Linear'|'Circular'|'Radial'|'Azimuthal'
        'e':     0.0,       # Normalised inner radius of annular aperture (0=full)
        'm':     0,         # Topological charge (0=plane wave)
        'plane': 'R-Z',     # Calculation plane: 'X-Y'|'R-Z'
    }
